core/src/main/java/org/springframework/security/authorization/method/PostProcessableAuthorizationDecision.java

```diff
@@ -0,0 +1,52 @@
+/*
+ * Copyright 2002-2024 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.springframework.security.authorization.method;
+
+import org.springframework.expression.Expression;
+import org.springframework.security.authorization.AuthorizationResult;
+import org.springframework.security.authorization.ExpressionAuthorizationDecision;
+import org.springframework.util.Assert;
+
+/**
+ * An {@link org.springframework.security.authorization.AuthorizationDecision} that post
+ * process an {@link AuthorizationResult} by invoking the
+ * {@link AuthorizationDeniedPostProcessor} passed via constructor.
+ *
+ * @param <T> the type of the object used in post-processing
+ * @author Marcus da Coregio
+ * @since 6.3
+ */
+class PostProcessableAuthorizationDecision<T> extends ExpressionAuthorizationDecision {
```

Comment:
I believe we don't need this anymore since we have `PostAuthorizeAuthorizationDecision` and `PreAuthorizeAuthorizationDecision`, both package-private.